### PR TITLE
scaffold nft module

### PR DESCRIPTION
**Pull request by lydia-pierce**
Scaffolded the cosmos nft module. Working on filling in the rest and testing.

### CODE REVIEW

File: x/nft/client/testutil/grpc.go [hunk 1]
```diff
@@ -0,0 +1,518 @@
+package testutil
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/testutil/rest"
```
Comment (lydia-pierce) [inline x/nft/client/testutil/grpc.go:6]: This is only in [v0.46.0-alpha1](https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.46.0-alpha1?tab=versions) and above of the cosmos-sdk and we need to use v0.44.3 everywhere else, so this is a bit problematic.

File: x/nft/simulation/operations_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	"github.com/stretchr/testify/suite"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/bank/testutil"
```
Comment (lydia-pierce) [inline x/nft/simulation/operations_test.go:17]: This is only in [v0.46.0-alpha1](https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.46.0-alpha1?tab=versions) and above of the cosmos-sdk and we need to use v0.44.3 everywhere else, so this is a bit problematic.

File: x/nft/keeper/keeper_test.go [hunk 3]
```diff
@@ -0,0 +1,363 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	tmtime "github.com/tendermint/tendermint/libs/time"
```
Comment (lydia-pierce) [inline x/nft/keeper/keeper_test.go:7]: We are using v0.34.14 of tendermint, however,`tendermint/libs/time` is only in [v0.35.1](https://pkg.go.dev/github.com/tendermint/tendermint?tab=versions) of tendermint

File: x/nft/client/testutil/grpc.go (hunk 1 above)
Comment (lydia-pierce) [reply x/nft/client/testutil/grpc.go:6]: should be solved with the protocgen.sh that uses third_party/proto

File: x/nft/simulation/operations_test.go (hunk 2 above)
Comment (lydia-pierce) [reply x/nft/simulation/operations_test.go:17]: should be solved with the protocgen.sh that uses third_party/proto

File: x/nft/keeper/keeper_test.go (hunk 3 above)
Comment (lydia-pierce) [reply x/nft/keeper/keeper_test.go:7]: should be solved with the protocgen.sh that uses third_party/proto

File: x/nft/types/expected_keepers.go [hunk 4]
```diff
@@ -0,0 +1 @@
+package types
```
Comment (evan-forbes) [inline x/nft/types/expected_keepers.go]: wut dis

**Review by evan-forbes — APPROVED** [review]
LFG LGTM :+1: :rocket: :100: 12,000 loc wcgw

**Comment by lydia-pierce** [comment]
This code was copied from the cosmos-sdk commit dd0d712. A `types` folder was added to the nft module, which was missing from the cosmos-sdk version of the nft module.

A `third_party/proto` folder was added to make use of remote proto files.

**Pull request merged**